build/arch.mk

```diff
@@ -29,3 +29,15 @@ ASMFLAGS += -I$(SRC_PATH)codec/common/arm64/
 CFLAGS += -DHAVE_NEON_AARCH64
 endif
 endif
+
+#for loongson
+ifneq ($(filter mips mips64, $(ARCH)),)
```

Comment:
the logic is correct, if ARCH is mips64 or mips, then $(filter mips mips64, $(ARCH)) is mips64 or mips64, so it is not equal to NULL.

Reply:
I think MMI compatible with MIPS. 
when someone wants to run a pure MIPS code without MMI, this "ifneq ($(filter mips mips64, $(ARCH)),)" branch will be include. this may cause an error. Is it better to separate this two architectures?

Reply:
Yes, i have considered how to separate those two architectures. Finally, i think use g++ predefined macro is an effective and simple method. The LOONGSON3A = $(shell g++ -dM -E - < /dev/null | grep '_MIPS_TUNE ' | cut -f 3 -d " ") gets the g++ predefined macro "_MIPS_TUNE". If the value is "loonson3a", the code will open mmi optimization. If not, the code is pure mips(but there is no mips assembly optimization so far) without mmi. So, it will not cause an error in other platform.

Reply:
maybe a solution.

Reply:
which version is your g++？I can not get "_MIPS_TUNE" on my local PC.

Reply:
in my PC, g++ -v:
![default](https://user-images.githubusercontent.com/41367273/45799018-a4e47f80-bcde-11e8-903a-6cfa8572d090.PNG)
if you are not in loongson or other mips platform, you will not find the "_MIPS_TUNE" in g++  predefined macro.

Reply:
Can these patches be merged?

Reply:
you' better separate MMI from MIPS code branch. mix this two code may cause pure MIPS code build error. thanks.

Reply:
I think use g++ predefined macro is an effective and simple method,  Can you accept this solution?